### PR TITLE
Add compressed reports for financial ledgers

### PR DESCRIPTION
**Pull request by mdietrichc2c**
New module: given that too many entries can make the general ledger/partners ledger reports fail due to lack of resources (SELECTs with too many move line ids, wkhtmltopdf lacking free memory space to generate the file, etc.), they can now be printed in "split" PDF files, all regrouped in a ZIP archive.

Rules for splitting files:
- partners ledger: one subfolder per account, max. 20000 ledger lines per account (a partner is added all at once). The files will be named according to the first and last partners in it.
- general ledger: one subfolder per account, 20000 ledger lines per file, opening balance included. Numbered 1-n at the end.
  For both reports, a "balance" file is also created at the root of the archive, since the Webkit reports depended on the lines present in it to compute it.

What's still needed:
- unit tests,
- probably create a new module 'report_zip', akin to 'report_webkit' and 'report_xls', allowing to create new reports more generically,
- any improvements to the report layout/file naming convention.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I see a possible misunderstanding in the names, because zip is also used por postal codes. Can you rename it to another thing like `_compressed` or something similar?

File: account_financial_report_compressed/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        per account. This is useful if the customer has many lines.
+    """,
+    'depends': ['account_financial_report_webkit'],
+    'demo_xml': [],
+    'init_xml': [],
+    'update_xml': [
+        'report.xml',
+        'wizard/general_ledger_wizard_view.xml',
+        'wizard/partners_ledger_wizard_view.xml',
+    ],
```
Comment (laetitia-gangloff) [inline account_financial_report_compressed/__openerp__.py:37]: demo_xml is replaced by demo.
init_xml and update_xml are replaced by data.

https://www.odoo.com/files/memento/OpenERP_Technical_Memento_latest.pdf

**Comment by gurneyalex** [comment]
@mdietrichc2c can you apply laetitia's request and rebase your PR?

**Comment by mdietrichc2c** [comment]
@gurneyalex @laetitia-gangloff done, thanks

File: account_financial_report_compressed/wizard/general_ledger_wizard_view.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+  <data>
+
+    <record id="general_ledger_compressed_wizard_view" model="ir.ui.view">
+      <field name="name">general.ledger.compressed</field>
+      <field name="model">general.ledger.webkit</field>
+      <field name="type">form</field>
```
Comment (gurneyalex) [inline account_financial_report_compressed/wizard/general_ledger_wizard_view.xml:8]: remove field `type` (deprecated on `ir.ui.view`)to make runbot green

File: account_financial_report_compressed/wizard/partners_ledger_wizard_view.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+  <data>
+
+      <record id="partners_ledger_compressed_wizard_view" model="ir.ui.view">
+          <field name="name">partners.ledger.compressed</field>
+          <field name="model">partners.ledger.webkit</field>
+          <field name="type">form</field>
```
Comment (gurneyalex) [inline account_financial_report_compressed/wizard/partners_ledger_wizard_view.xml:8]: remove field `type` (deprecated on `ir.ui.view`)to make runbot green

File: account_financial_report_compressed/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#
+##############################################################################
+{
+    'name': 'Add multiple export to ledger reports',
+    'version': '0.1',
+    'license': 'AGPL-3',
+    'author': 'Camptocamp',
+    'category': 'Generic Modules/Accounting',
+    'description': """
+        This module add a multiple export option for ledgers
```
Comment (gurneyalex) [inline account_financial_report_compressed/__openerp__.py:28]: can you extract the description to README.rst with OCA template? 

s/add/adds/

File: account_financial_report_compressed/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'Add multiple export to ledger reports',
+    'version': '0.1',
+    'license': 'AGPL-3',
+    'author': 'Camptocamp',
```
Comment (gurneyalex) [inline account_financial_report_compressed/__openerp__.py:25]: please add 'Odoo Community Association (OCA)'

File: account_financial_report_compressed/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    'category': 'Generic Modules/Accounting',
+    'description': """
+        This module add a multiple export option for ledgers
+        (partners and general), in order to export one PDF file
+        per account. This is useful if the customer has many lines.
```
Comment (gurneyalex) [inline account_financial_report_compressed/__openerp__.py:30]: which customer?

**Comment by gurneyalex** [comment]
I'm concerned by the large chunk of untested code. Could you add an automated test?

**Comment by ovnicraft** [comment]
@mdietrichc2c any update here ?

**Comment by mdietrichc2c** [comment]
@ovnicraft no, sorry ; this PR is pretty much deprecated now, and the future QWeb reports work without any memory issues. I am closing this PR.

**Pull request closed without merging**